build.rs

```diff
@@ -30,18 +30,91 @@ use std::{
 mod env {
     use std::ffi::OsString;
 
-    /// Read an environment variable and tell Cargo that we depend on it.
+    enum EnvVarTy {
+        RerunIfChanged,
+        SetByCargo,
+    }
+
+    pub struct EnvVar {
+        name: &'static str,
+        ty: EnvVarTy,
+    }
+
+    /// Read an environment variable and optionally tell Cargo that we depend on it.
     ///
-    /// The name is static since we intend to only read a static set of environment
+    /// The env var is static since we intend to only read a static set of environment
     /// variables.
-    pub fn var_os(name: &'static str) -> Option<OsString> {
-        println!("cargo:rerun-if-env-changed={}", name);
-        std::env::var_os(name)
+    pub fn var_os(env_var: &'static EnvVar) -> Option<OsString> {
+        if let EnvVarTy::SetByCargo = env_var.ty {
+            println!("cargo:rerun-if-env-changed={}", env_var.name);
+        }
+        std::env::var_os(env_var.name)
     }
 
-    pub fn var(name: &'static str) -> Option<String> {
-        var_os(name).and_then(|value| value.into_string().ok())
+    pub fn var(env_var: &'static EnvVar) -> Option<String> {
+        var_os(env_var).and_then(|value| value.into_string().ok())
     }
+
+    pub const CARGO_MANIFEST_DIR: EnvVar = EnvVar {
```

Comment:
I guess we need to decide some kind of order for these. I suggest alphabetical order. Could you alphabetize them and then add a comment above the first one `// In alphabetical order` or similar?

Reply:
Done.